apps/mobile/src/screens/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,205 @@
+} from "./libs/constants/constants";
+import {
+	type OnboardingAnswerRequestBodyDto,
+	type OnboardingFormValues,
+} from "./libs/types/types";
+import { styles } from "./styles";
+
+type RootStackParameterList = {
+	[RootScreenName.WELCOME]: undefined;
+};
+
+const ZERO = 0;
+
+const Onboarding: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const pagerReference = useRef<InfinitePagerImperativeApi | null>(null);
+
+	const navigation =
+		useNavigation<NativeStackNavigationProp<RootStackParameterList>>();
+
+	const {
+		allAnswers,
+		currentAnswer,
+		currentQuestionIndex,
+		dataStatus,
+		isLastQuestion,
+		question,
+		totalQuestionsAmount,
+	} = useAppSelector(({ onboarding }) => ({
+		allAnswers: onboarding.answersByQuestionIndex,
+		currentAnswer:
+			onboarding.answersByQuestionIndex[onboarding.currentQuestionIndex] || "",
+		currentQuestionIndex: onboarding.currentQuestionIndex,
+		dataStatus: onboarding.dataStatus,
+		isLastQuestion:
+			onboarding.currentQuestionIndex ===
+			onboarding.questions.length - PREVIOUS_INDEX_OFFSET,
+		question: onboarding.currentQuestion,
+		totalQuestionsAmount: onboarding.questions.length - PREVIOUS_INDEX_OFFSET,
+	}));
+
+	useEffect(() => {
+		void dispatch(onboardingActions.getAll());
+	}, [dispatch]);
+
+	const { control, errors, handleSubmit, isValid, reset } =
+		useAppForm<OnboardingFormValues>({
+			defaultValues: ONBOARDING_FORM_DEFAULT_VALUES,
+			validationSchema: oneAnswerSelectedValidationSchema,
+		});
+
+	const handleSaveAnswers = useCallback(
+		(payload: OnboardingAnswerRequestBodyDto) => {
+			//TODO: save data to backend
+			return payload;
+		},
+		[],
+	);
+
+	useEffect(() => {
+		reset({ answer: currentAnswer.toString() });
+	}, [currentQuestionIndex, currentAnswer, reset]);
+
+	const handleNextClick = useCallback(
+		(payload: OnboardingFormValues) => {
+			dispatch(
+				onboardingActions.setAnswersByQuestionIndex({
+					answerId: Number(payload.answer),
+					questionIndex: currentQuestionIndex,
+				}),
+			);
+
+			if (isLastQuestion) {
+				handleSaveAnswers({ answerIds: allAnswers });
+
+				return;
+			}
+
+			if (pagerReference.current) {
+				pagerReference.current.incrementPage({ animated: true });
+			}
+
+			void dispatch(onboardingActions.nextQuestion());
+		},
+
+		[
+			isLastQuestion,
+			dispatch,
+			handleSaveAnswers,
+			currentQuestionIndex,
+			allAnswers,
+		],
+	);
+
+	const handlePreviousClick = useCallback(() => {
+		void dispatch(onboardingActions.previousQuestion());
+
+		if (pagerReference.current) {
+			pagerReference.current.decrementPage({ animated: true });
+		}
+	}, [dispatch]);
+
+	const handleFormSubmit = useCallback((): void => {
+		void handleSubmit(handleNextClick)();
+	}, [handleNextClick, handleSubmit]);
+
+	const handleAnalyzePress = useCallback((): void => {
+		void handleSubmit(handleNextClick)();
+		navigation.navigate(RootScreenName.WELCOME);
+	}, [navigation, handleNextClick, handleSubmit]);
```

Comment:
Maybe let's simplify it a little and then we don't need ternary in `onPress` below
```suggestion
	const handleFormSubmit = useCallback((): void => {
		void handleSubmit(handleNextClick)();
		
		if (isLastQuestion) {
		    navigation.navigate(RootScreenName.WELCOME);
		}
	}, [handleNextClick, handleSubmit, isLastQuestion]);
```